Duplicati/Library/Backend/SSHv2/SSHv2Backend.cs

```diff
@@ -328,11 +328,44 @@ private void CreateConnection()
             if (m_keepaliveinterval.Ticks != 0)
                 con.KeepAliveInterval = m_keepaliveinterval;
 
-            con.Connect();
+            this.TryConnect(con);
 
             m_con = con;
         }
 
+        private void TryConnect(SftpClient client)
+        {
+            try
+            {
+                client.Connect();
+            }
+            catch (NotImplementedException)
+            {
+                // SSH.NET relies on the System.Security.Cryptography.ECDsaCng class for
+                // ECDSA algorithms, which is not implemented in Mono (as of 6.12.0.144).
+                // This prevents clients from connecting if one of the ECDSA algorithms is
+                // chosen as the host key algorithm.  In the event that this causes a
+                // connection failure, we will prevent the client from advertising support
+                // for ECDSA algorithms and make another connection attempt.
+                //
+                // See https://github.com/mono/mono/blob/mono-6.12.0.144/mcs/class/referencesource/System.Core/System/Security/Cryptography/ECDsaCng.cs
+                if (Utility.Utility.IsMono)
+                {
+                    IEnumerable<string> ecdsaKeys = client.ConnectionInfo.HostKeyAlgorithms.Keys.Where(x => x.StartsWith("ecdsa"));
+                    foreach (string key in ecdsaKeys)

```

Comment:
I had to change `ecdsaKeys` to `ecdsaKeys.ToList()` otherwise this exception was getting thrown:

`System.InvalidOperationException: Collection was modified; enumeration operation may not execute.`

Not sure if adding the `.ToList()` is the best way to solve this but it works.

Reply:
Thanks for testing and noticing this.  That makes sense, we need to enumerate the items before modifying the collection.  I fixed this in revision b80845a17652fe333145d90bc2509a732d02a517.  Would you mind testing again, just to be safe?

Reply:
Just tested...works great!